run.py

```diff
@@ -219,225 +317,603 @@ def run_diffusion_processsing(**args):
-        t1_zooms = nibabel.load(t1ws[0]).get_header().get_zooms()
-        t1_res = float(min(t1_zooms[:3]))
-        t1_template_res = min(available_resolutions, key=lambda x:abs(float(x)-t1_res))
-        t2_zooms = nibabel.load(t2ws[0]).get_header().get_zooms()
-        t2_res = float(min(t2_zooms[:3]))
-        t2_template_res = min(available_resolutions, key=lambda x:abs(float(x)-t2_res))
-
-        fieldmap_set = layout.get_fieldmap(t1ws[0])
-        fmap_args = {"fmapmag": "NONE",
-                     "fmapphase": "NONE",
-                     "echodiff": "NONE",
-                     "t1samplespacing": "NONE",
-                     "t2samplespacing": "NONE",
-                     "unwarpdir": "NONE",
-                     "avgrdcmethod": "NONE",
-                     "SEPhaseNeg": "NONE",
-                     "SEPhasePos": "NONE",
-                     "echospacing": "NONE",
-                     "seunwarpdir": "NONE"}
-
-        if fieldmap_set:
-            t1_spacing = layout.get_metadata(t1ws[0])["EffectiveEchoSpacing"]
-            t2_spacing = layout.get_metadata(t2ws[0])["EffectiveEchoSpacing"]
-
-            unwarpdir = layout.get_metadata(t1ws[0])["PhaseEncodingDirection"]
-            unwarpdir = unwarpdir.replace("i","x").replace("j", "y").replace("k", "z")
-            if len(unwarpdir) == 2:
-                unwarpdir = unwarpdir[0]+"-"
-
-            fmap_args.update({"t1samplespacing": "%.8f"%t1_spacing,
-                              "t2samplespacing": "%.8f"%t2_spacing,
-                              "unwarpdir": unwarpdir})
-
-            if fieldmap_set["type"] == "phasediff":
-                merged_file = "%s/tmp/%s/magfile.nii.gz"%(args.output_dir, subject_label)
-                run("mkdir -p %s/tmp/%s/ && fslmerge -t %s %s %s"%(args.output_dir,
-                subject_label,
-                merged_file,
-                fieldmap_set["magnitude1"],
-                fieldmap_set["magnitude2"]))
-
-                phasediff_metadata = layout.get_metadata(fieldmap_set["phasediff"])
-                te_diff = phasediff_metadata["EchoTime2"] - phasediff_metadata["EchoTime1"]
-                # HCP expects TE in miliseconds
-                te_diff = te_diff*1000.0
-
-                fmap_args.update({"fmapmag": merged_file,
-                                  "fmapphase": fieldmap_set["phasediff"],
-                                  "echodiff": "%.6f"%te_diff,
-                                  "avgrdcmethod": "SiemensFieldMap"})
-            elif fieldmap_set["type"] == "epi":
-                SEPhaseNeg = None
-                SEPhasePos = None
-                for fieldmap in fieldmap_set["epi"]:
-                    enc_dir = layout.get_metadata(fieldmap)["PhaseEncodingDirection"]
-                    if "-" in enc_dir:
-                        SEPhaseNeg = fieldmap
+        # before do anything else add IntendedFor field to fieldmap
+        setup(os.path.join(args.bids_dir, "sub-"+subject_label))
```

Comment:
Modifying input dataset in place is a big red flag. This bahaviour will not be expected by users and can cause a lot of trouble. We actually recommend to pass  `bids_dir` as a read-only volume which would break this. 

Handling of missing metadata (such as IntendedFor) should be done before running BIDS Apps.